### PR TITLE
Support for UUID/uniqueidentifier data types in MSSQL.

### PR DESCRIPTION
**Pull request by Yaytay**
Motivation:

Support for UUID/uniqueidentifier data types in MSSQL.
Our databases make a lot of use of uniqueidentifier columns - without support for them we cannot use vertx-sql-client.

### CODE REVIEW

**Comment by Yaytay** [comment]
Just noticed this: https://github.com/eclipse-vertx/vertx-sql-client/issues/965
I can reduce from four ByteBuf:write* calls to two writeLong calls or even one writeBytes call if I do the byte-twiddling myself.
Let me know if you'd like me to make that change.

**Comment by vietj** [comment]
thanks for the contribution @Yaytay

File: vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java [hunk 1]
```diff
@@ -240,7 +241,56 @@ public Object decodeValue(ByteBuf byteBuf, Metadata metadata) {
...
+    @Override
+    public Metadata decodeMetadata(ByteBuf byteBuf) {
+      Metadata metadata = new Metadata();
+      metadata.length = byteBuf.readByte();
+      return metadata;
+    }
+
+    @Override
+    public JDBCType jdbcType(Metadata metadata) {
+      if (metadata.length == 16) return JDBCType.CHAR;
```
Comment (tsegismont) [inline vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java:254]:
```suggestion
      if (metadata.length == 16) return JDBCType.GUID;
```

**Review by tsegismont — CHANGES_REQUESTED** [review]
Thank you @Yaytay , great job!

Just a couple changes needed. Do you think you can do it before tomorrow morning CEST ? We'll release a new 4.2 version and, in this case, the change will be part of it.

File: vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java (hunk 1 above)
Comment (Yaytay) [reply vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java:254]: There is no JDBCType.GUID in https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html.
Any other idea?
JDBCType.OTHER?
JDBCType.OBJECT?
JDBCType.BINARY?

**Review by tsegismont — APPROVED** [review]
Thanks @Yaytay 

I'll go ahead and merge so that the PR is included in 4.2.3

File: vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java (hunk 1 above)
Comment (tsegismont) [reply vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/DataType.java:254]: Changed it to `OTHER`

**Pull request merged**